### PR TITLE
Add cluster definition for the dishwasher mode select cluster.

### PR DESCRIPTION
**Pull request by rbultman**
This PR adds the cluster definition for the dishwasher mode select cluster.

1. Adds the cluster definition XML file to src/app/zap-templates/zcl/data-model/chip
2. Updates scritps/rules.matterlint
3. Updates src/app/zap/zap-templates/zcl/data-model/all.xml
4. Updates src/app/zap-templates/zcl/zcl-with-test-extensions.json
5. Updates src/app/zap-templates/zcl/zcl.json
6. Updates src/controller/data_model/controller-clusters.zap
7. Updates src/app/zap_cluster_list.json

### CODE REVIEW

File: src/app/zap-templates/zcl/zcl.json
Comment (p0fi) [inline src/app/zap-templates/zcl/zcl.json:1]: The many unnecessary formatting changes in this file should be taken out I think.

File: src/app/zap_cluster_list.json
Comment (p0fi) [inline src/app/zap_cluster_list.json:1]: Same thing: many unnecessary formatting changes

File: .github/workflows/tests.yaml
Comment (p0fi) [inline .github/workflows/tests.yaml:1]: Huge amount of formatting changes

File: src/app/zap-templates/zcl/data-model/chip/dishwasher-mode-select-cluster.xml
Comment (p0fi) [inline src/app/zap-templates/zcl/data-model/chip/dishwasher-mode-select-cluster.xml:1]: This should not have functional influence but fwiw there are other files which place the enums below the cluster definition. I think a golden standard template for this kind of file is missing.

**Review by p0fi — CHANGES_REQUESTED** [review]
I think the PR description and title could use a little love. They could be a bit clearer on what are the indented changes and what was the process followed as for me. 

Also please check your toolchain if it producing so many formatting changes something seems a bit off.

**Comment by rbultman** [comment]
> I think the PR description and title could use a little love. They could be a bit clearer on what are the indented changes and what was the process followed as for me.
> 
> Also please check your toolchain if it producing so many formatting changes something seems a bit off.

Where would I check for this?  Perhaps I am missing some plugins?

File: src/app/zap-templates/zcl/zcl-with-test-extensions.json [hunk 1]
```diff
@@ -96,9 +97,15 @@
     "manufacturersXml": "../../../../src/app/zap-templates/zcl/data-model/manufacturers.xml",
     "options": {
         "text": {
-            "defaultResponsePolicy": ["Always", "Conditional", "Never"]
+            "defaultResponsePolicy": [
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/zcl-with-test-extensions.json:100]: Again, looks like you have some sort of auto-formatting turned on that disagrees with some existing auto-formatting.

File: src/app/zap_cluster_list.json
Comment (bzbarsky-apple) [inline src/app/zap_cluster_list.json:1]: And the auto-formatting disagreeing with the restyled automation _here_ makes this PR fail CI.

File: src/app/zap-templates/zcl/zcl.json
Comment (rbultman) [reply src/app/zap-templates/zcl/zcl.json:1]: Corrected.

File: .github/workflows/tests.yaml
[the remark this answers is not in the document]
Comment (rbultman) [reply .github/workflows/tests.yaml:1]: Corrected.

File: src/app/zap-templates/zcl/zcl-with-test-extensions.json (hunk 1 above)
Comment (rbultman) [reply src/app/zap-templates/zcl/zcl-with-test-extensions.json]: Corrected.

File: src/app/zap-templates/zcl/data-model/chip/dishwasher-mode-select-cluster.xml [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+      <arg name="Status" type="ENUM8" />
+      <arg name="StatusText" type="CHAR_STRING" />
+    </command>
+
+  </cluster>
+
+  <bitmap name="DishwasherModeSelectFeature" type="BITMAP32">
+    <cluster code="0x0059"/>
+    <field name="DEPONOFF" mask="0x1"/>
+    <field name="EXT" mask="0x1"/>
```
[the remark this answers is not in the document]
Comment (bzbarsky-apple) [reply src/app/zap-templates/zcl/data-model/chip/dishwasher-mode-select-cluster.xml:99]: It looks like https://github.com/project-chip/connectedhomeip/pull/26508 actually includes this cluster @rbultman ?

**Comment by rbultman** [comment]
This XML work is already covered in PR #26508, so this PR is being closed.
